### PR TITLE
Use esConsumes in some TauAnalysis/MCEmbeddingTools EDModules

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

Part of #31061.

#### PR validation:

Unit tests of `TauAnalysis/MCEmbeddingTools` succeed when run with #35588.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: TauAnalysis/MCEmbeddingTools/plugins/EmbeddingVertexCorrector.cc [hunk 1]
```diff
@@ -89,15 +87,6 @@ EmbeddingVertexCorrector::~EmbeddingVertexCorrector() {}
...
 //
 
-void EmbeddingVertexCorrector::beginRun(const edm::Run&, const edm::EventSetup& iEventSetup) {
-  // edm::ESHandle< SimBeamSpotObjects > beamhandle;
-  // iEventSetup.get<SimBeamSpotObjectsRcd>().get(beamhandle);
-
-  edm::ESHandle<BeamSpotObjects> beamhandle;
-  iEventSetup.get<BeamSpotObjectsRcd>().get(beamhandle);
-
-  edm::LogInfo("TauEmbedding") << "beam handle\n" << (*beamhandle);
```
Comment (makortel) [inline TauAnalysis/MCEmbeddingTools/plugins/EmbeddingVertexCorrector.cc:99]: Consuming an ESProduct just for a INFO-level printout seemed potentially unnecessary to me, so I'm suggesting to remove it. If the printout is deemed necessary, I'll change the PR accordingly.

**Comment by makortel** [comment]
@cmsbuild, please test

Failure seems to be caused by timeout.

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**